core/services/ocr2/plugins/ocr2vrf/coordinator/interfaces.go

```diff
@@ -11,7 +11,7 @@ import (
 
 //go:generate mockery --name VRFBeaconCoordinator --output ./mocks/ --case=underscore
 
-// VRFBeaconCoordinator is a narrow interface implemented by the contract go wrappers.
+// VRFBeaconCoordinator is an interface that defines methods needed by the off-chain coordinator
 type VRFBeaconCoordinator interface {
```

Comment:
Should this be renamed to just `VRFBeacon` or `VRFCoordinator`? Or are these methods in the union of `VRFBeacon` and `VRFCoordinator`?

Reply:
Ah I see, so this seems to be the latter, some are implemented by `VRFCoordinator`, some by `VRFBeacon`. I wonder if it's better to split this into two interfaces then, lest things get really confusing. Ideally our off-chain abstractions of on-chain code should mirror what is on-chain.

Reply:
It seems like `VRFCoordinator` implements `IBeaconPeriodBlocks` and `GetConfirmationDelays`, and `VRFBeacon` implements `SProvingKeyHash` and `SKeyID`, and both have a `ParseLog` implementation, which I can see would be convenient to have in one place rather than separate implementations.

Maybe if we're leaning towards this approach we should comment in the interface clearly which methods belong to which on-chain contract, that would make things much clearer.

Reply:
yes comments will help here. Added comments to `VRFRouter` instead. 

One tangetial comment: If we decide to have the third proxy contracts that routes calls to VRF Beacon and VRF Coordinator, we may only need to interact with one contract. Need confirm how geth wrappers will look for the proxy contract though.

Reply:
> Need confirm how geth wrappers will look for the proxy contract though.

There won't be a specific gethwrapper for the proxy contract - we'll have to use the gethwrapper of the logic contract that is being proxied, but give it the address of the proxy contract and not the logic contract. Pseudocode:

```golang
// proxy is at 0xA
// logic is at 0xB
wrapper, err := logic_contract_gethwrapper.NewLogicContract(proxyAddress, backend)
if err != nil {
 // ... 
}
wrapper.Function(...) // call func on proxy using ABI of the logic contract
```